### PR TITLE
Remove auto labeling "ambient-type"= "workload"

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@howardjohn Not familiar with the ambient integration test 

https://github.com/istio/istio/blob/4c1097808cd1972f541400ce240191ebb8ad2481/tests/integration/ambient/main_test.go#L201

Maybe need to remove some test as well

File: pilot/pkg/ambient/ambientpod/podutil.go [hunk 1]
```diff
@@ -92,10 +92,7 @@ func ShouldPodBeInIpset(namespace *corev1.Namespace, pod *corev1.Pod, meshMode s
 
 // @TODO Interim function for waypoint proxy, to be replaced after design meeting
 func PodHasOptOut(pod *corev1.Pod) bool {
-	if val, ok := pod.Labels["ambient-type"]; ok {
-		return val == "waypoint" || val == "none"
```
Comment (hzxuzhonghu) [inline pilot/pkg/ambient/ambientpod/podutil.go:96]: suspect this is related to test failure

**Comment by howardjohn** [comment]
I was working on similar as part of https://github.com/istio/istio/issues/43254. will send a PR

**Comment by howardjohn** [comment]
https://github.com/istio/istio/pull/43313

**Comment by hzxuzhonghu** [comment]
Close in favor of that

**Pull request closed without merging**